storehaus-kafka/src/main/scala/com/twitter/storehaus/kafka/KafkaInjections.scala

```diff
@@ -16,40 +16,48 @@
 
 package com.twitter.storehaus.kafka
 
-import kafka.serializer.{Decoder, Encoder}
+import java.util
+
 import com.twitter.bijection.{Codec, Injection}
+import org.apache.kafka.common.serialization.{Serializer, Deserializer}
 
 /**
- * @author Mansur Ashraf
- * @since 11/23/13
- */
-object KafkaInjections {
+  * @author Mansur Ashraf
+  * @since 11/23/13
+  */
+private[kafka] object KafkaInjections {
 
   class ByteArrayEncoder extends FromInjectionEncoder[Array[Byte]] {
     def injection: Injection[Array[Byte], Array[Byte]] = Injection.identity[Array[Byte]]
   }
 
-  trait FromInjectionDecoder[T] extends Decoder[T] {
+  trait FromInjectionDecoder[T] extends Deserializer[T] with SerDe {
     def injection: Injection[T, Array[Byte]]
 
-    override def fromBytes(bytes: Array[Byte]): T = injection.invert(bytes).get
+    override def deserialize(topic: String, data: Array[Byte]): T =
+      injection.invert(data).get
   }
 
-  trait FromInjectionEncoder[T] extends Encoder[T] {
+  trait FromInjectionEncoder[T] extends Serializer[T] with SerDe {
     def injection: Injection[T, Array[Byte]]
 
-    override def toBytes(t: T): Array[Byte] = injection(t)
+    override def serialize(topic: String, data: T): Array[Byte] =
+      injection(data)
   }
 
-  def fromInjection[T: Codec]: (Encoder[T], Decoder[T]) = {
+  trait SerDe {
+    def configure(configs: util.Map[String, _], isKey: Boolean): Unit = ()
+    def close(): Unit = ()
+  }
+
+  def fromInjection[T: Codec]: (Serializer[T], Deserializer[T]) = {
```

Comment:
how do these get used if below? It seems kafka creates the serializers with reflection.